// portgas-d-ui/dist/portgas-d-ui.es.js
import Te, { useState as we } from "react";
var C = { exports: {} }, w = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var fe;
function ke() {
  if (fe) return w;
  fe = 1;
  var s = Symbol.for("react.transitional.element"), b = Symbol.for("react.fragment");
  function g(p, i, f) {
    var m = null;
    if (f !== void 0 && (m = "" + f), i.key !== void 0 && (m = "" + i.key), "key" in i) {
      f = {};
      for (var j in i)
        j !== "key" && (f[j] = i[j]);
    } else f = i;
    return i = f.ref, {
      $$typeof: s,
      type: p,
      key: m,
      ref: i !== void 0 ? i : null,
      props: f
    };
  }
  return w.Fragment = b, w.jsx = g, w.jsxs = g, w;
}
var k = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var de;
function Re() {
  return de || (de = 1, process.env.NODE_ENV !== "production" && function() {
    function s(e) {
      if (e == null) return null;
      if (typeof e == "function")
        return e.$$typeof === je ? null : e.displayName || e.name || null;
      if (typeof e == "string") return e;
      switch (e) {
        case Y:
          return "Fragment";
        case ye:
          return "Portal";
        case I:
          return "Profiler";
        case F:
          return "StrictMode";
        case $:
          return "Suspense";
        case W:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (typeof e.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), e.$$typeof) {
          case Z:
            return (e.displayName || "Context") + ".Provider";
          case L:
            return (e._context.displayName || "Context") + ".Consumer";
          case M:
            var r = e.render;
            return e = e.displayName, e || (e = r.displayName || r.name || "", e = e !== "" ? "ForwardRef(" + e + ")" : "ForwardRef"), e;
          case U:
            return r = e.displayName || null, r !== null ? r : s(e.type) || "Memo";
          case q:
            r = e._payload, e = e._init;
            try {
              return s(e(r));
            } catch {
            }
        }
      return null;
    }
    function b(e) {
      return "" + e;
    }
    function g(e) {
      try {
        b(e);
        var r = !1;
      } catch {
        r = !0;
      }
      if (r) {
        r = console;
        var t = r.error, n = typeof Symbol == "function" && Symbol.toStringTag && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t.call(
          r,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          n
        ), b(e);
      }
    }
    function p() {
    }
    function i() {
      if (_ === 0) {
        D = console.log, ee = console.info, re = console.warn, te = console.error, oe = console.group, ne = console.groupCollapsed, ae = console.groupEnd;
        var e = {
          configurable: !0,
          enumerable: !0,
          value: p,
          writable: !0
        };
        Object.defineProperties(console, {
          info: e,
          log: e,
          warn: e,
          error: e,
          group: e,
          groupCollapsed: e,
          groupEnd: e
        });
      }
      _++;
    }
    function f() {
      if (_--, _ === 0) {
        var e = { configurable: !0, enumerable: !0, writable: !0 };
        Object.defineProperties(console, {
          log: h({}, e, { value: D }),
          info: h({}, e, { value: ee }),
          warn: h({}, e, { value: re }),
          error: h({}, e, { value: te }),
          group: h({}, e, { value: oe }),
          groupCollapsed: h({}, e, { value: ne }),
          groupEnd: h({}, e, { value: ae })
        });
      }
      0 > _ && console.error(
        "disabledDepth fell below zero. This is a bug in React. Please file an issue."
      );
    }
    function m(e) {
      if (z === void 0)
        try {
          throw Error();
        } catch (t) {
          var r = t.stack.trim().match(/\n( *(at )?)/);
          z = r && r[1] || "", le = -1 < t.stack.indexOf(`
    at`) ? " (<anonymous>)" : -1 < t.stack.indexOf("@") ? "@unknown:0:0" : "";
        }
      return `
` + z + e + le;
    }
    function j(e, r) {
      if (!e || H) return "";
      var t = V.get(e);
      if (t !== void 0) return t;
      H = !0, t = Error.prepareStackTrace, Error.prepareStackTrace = void 0;
      var n = null;
      n = x.H, x.H = null, i();
      try {
        var u = {
          DetermineComponentFrameRoot: function() {
            try {
              if (r) {
                var E = function() {
                  throw Error();
                };
                if (Object.defineProperty(E.prototype, "props", {
                  set: function() {
                    throw Error();
                  }
                }), typeof Reflect == "object" && Reflect.construct) {
                  try {
                    Reflect.construct(E, []);
                  } catch (v) {
                    var R = v;
                  }
                  Reflect.construct(e, [], E);
                } else {
                  try {
                    E.call();
                  } catch (v) {
                    R = v;
                  }
                  e.call(E.prototype);
                }
              } else {
                try {
                  throw Error();
                } catch (v) {
                  R = v;
                }
                (E = e()) && typeof E.catch == "function" && E.catch(function() {
                });
              }
            } catch (v) {
              if (v && R && typeof v.stack == "string")
                return [v.stack, R.stack];
            }
            return [null, null];
          }
        };
        u.DetermineComponentFrameRoot.displayName = "DetermineComponentFrameRoot";
        var a = Object.getOwnPropertyDescriptor(
          u.DetermineComponentFrameRoot,
          "name"
        );
        a && a.configurable && Object.defineProperty(
          u.DetermineComponentFrameRoot,
          "name",
          { value: "DetermineComponentFrameRoot" }
        );
        var o = u.DetermineComponentFrameRoot(), d = o[0], N = o[1];
        if (d && N) {
          var c = d.split(`
`), y = N.split(`
`);
          for (o = a = 0; a < c.length && !c[a].includes(
            "DetermineComponentFrameRoot"
          ); )
            a++;
          for (; o < y.length && !y[o].includes(
            "DetermineComponentFrameRoot"
          ); )
            o++;
          if (a === c.length || o === y.length)
            for (a = c.length - 1, o = y.length - 1; 1 <= a && 0 <= o && c[a] !== y[o]; )
              o--;
          for (; 1 <= a && 0 <= o; a--, o--)
            if (c[a] !== y[o]) {
              if (a !== 1 || o !== 1)
                do
                  if (a--, o--, 0 > o || c[a] !== y[o]) {
                    var T = `
` + c[a].replace(
                      " at new ",
                      " at "
                    );
                    return e.displayName && T.includes("<anonymous>") && (T = T.replace("<anonymous>", e.displayName)), typeof e == "function" && V.set(e, T), T;
                  }
                while (1 <= a && 0 <= o);
              break;
            }
        }
      } finally {
        H = !1, x.H = n, f(), Error.prepareStackTrace = t;
      }
      return c = (c = e ? e.displayName || e.name : "") ? m(c) : "", typeof e == "function" && V.set(e, c), c;
    }
    function S(e) {
      if (e == null) return "";
      if (typeof e == "function") {
        var r = e.prototype;
        return j(
          e,
          !(!r || !r.isReactComponent)
        );
      }
      if (typeof e == "string") return m(e);
      switch (e) {
        case $:
          return m("Suspense");
        case W:
          return m("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case M:
            return e = j(e.render, !1), e;
          case U:
            return S(e.type);
          case q:
            r = e._payload, e = e._init;
            try {
              return S(e(r));
            } catch {
            }
        }
      return "";
    }
    function A() {
      var e = x.A;
      return e === null ? null : e.getOwner();
    }
    function be(e) {
      if (K.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning) return !1;
      }
      return e.key !== void 0;
    }
    function me(e, r) {
      function t() {
        se || (se = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          r
        ));
      }
      t.isReactWarning = !0, Object.defineProperty(e, "key", {
        get: t,
        configurable: !0
      });
    }
    function Ee() {
      var e = s(this.type);
      return ue[e] || (ue[e] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), e = this.props.ref, e !== void 0 ? e : null;
    }
    function ge(e, r, t, n, u, a) {
      return t = a.ref, e = {
        $$typeof: P,
        type: e,
        key: r,
        props: a,
        _owner: u
      }, (t !== void 0 ? t : null) !== null ? Object.defineProperty(e, "ref", {
        enumerable: !1,
        get: Ee
      }) : Object.defineProperty(e, "ref", { enumerable: !1, value: null }), e._store = {}, Object.defineProperty(e._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(e, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.freeze && (Object.freeze(e.props), Object.freeze(e)), e;
    }
    function G(e, r, t, n, u, a) {
      if (typeof e == "string" || typeof e == "function" || e === Y || e === I || e === F || e === $ || e === W || e === pe || typeof e == "object" && e !== null && (e.$$typeof === q || e.$$typeof === U || e.$$typeof === Z || e.$$typeof === L || e.$$typeof === M || e.$$typeof === Ne || e.getModuleId !== void 0)) {
        var o = r.children;
        if (o !== void 0)
          if (n)
            if (J(o)) {
              for (n = 0; n < o.length; n++)
                B(o[n], e);
              Object.freeze && Object.freeze(o);
            } else
              console.error(
                "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
              );
          else B(o, e);
      } else
        o = "", (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (o += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports."), e === null ? n = "null" : J(e) ? n = "array" : e !== void 0 && e.$$typeof === P ? (n = "<" + (s(e.type) || "Unknown") + " />", o = " Did you accidentally export a JSX literal instead of a component?") : n = typeof e, console.error(
          "React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s",
          n,
          o
        );
      if (K.call(r, "key")) {
        o = s(e);
        var d = Object.keys(r).filter(function(c) {
          return c !== "key";
        });
        n = 0 < d.length ? "{key: someKey, " + d.join(": ..., ") + ": ...}" : "{key: someKey}", ce[o + n] || (d = 0 < d.length ? "{" + d.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          n,
          o,
          d,
          o
        ), ce[o + n] = !0);
      }
      if (o = null, t !== void 0 && (g(t), o = "" + t), be(r) && (g(r.key), o = "" + r.key), "key" in r) {
        t = {};
        for (var N in r)
          N !== "key" && (t[N] = r[N]);
      } else t = r;
      return o && me(
        t,
        typeof e == "function" ? e.displayName || e.name || "Unknown" : e
      ), ge(e, o, a, u, A(), t);
    }
    function B(e, r) {
      if (typeof e == "object" && e && e.$$typeof !== _e) {
        if (J(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            O(n) && X(n, r);
          }
        else if (O(e))
          e._store && (e._store.validated = 1);
        else if (e === null || typeof e != "object" ? t = null : (t = Q && e[Q] || e["@@iterator"], t = typeof t == "function" ? t : null), typeof t == "function" && t !== e.entries && (t = t.call(e), t !== e))
          for (; !(e = t.next()).done; )
            O(e.value) && X(e.value, r);
      }
    }
    function O(e) {
      return typeof e == "object" && e !== null && e.$$typeof === P;
    }
    function X(e, r) {
      if (e._store && !e._store.validated && e.key == null && (e._store.validated = 1, r = xe(r), !ie[r])) {
        ie[r] = !0;
        var t = "";
        e && e._owner != null && e._owner !== A() && (t = null, typeof e._owner.tag == "number" ? t = s(e._owner.type) : typeof e._owner.name == "string" && (t = e._owner.name), t = " It was passed a child from " + t + ".");
        var n = x.getCurrentStack;
        x.getCurrentStack = function() {
          var u = S(e.type);
          return n && (u += n() || ""), u;
        }, console.error(
          'Each child in a list should have a unique "key" prop.%s%s See https://react.dev/link/warning-keys for more information.',
          r,
          t
        ), x.getCurrentStack = n;
      }
    }
    function xe(e) {
      var r = "", t = A();
      return t && (t = s(t.type)) && (r = `

Check the render method of \`` + t + "`."), r || (e = s(e)) && (r = `

Check the top-level render call using <` + e + ">."), r;
    }
    var he = Te, P = Symbol.for("react.transitional.element"), ye = Symbol.for("react.portal"), Y = Symbol.for("react.fragment"), F = Symbol.for("react.strict_mode"), I = Symbol.for("react.profiler"), L = Symbol.for("react.consumer"), Z = Symbol.for("react.context"), M = Symbol.for("react.forward_ref"), $ = Symbol.for("react.suspense"), W = Symbol.for("react.suspense_list"), U = Symbol.for("react.memo"), q = Symbol.for("react.lazy"), pe = Symbol.for("react.offscreen"), Q = Symbol.iterator, je = Symbol.for("react.client.reference"), x = he.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, K = Object.prototype.hasOwnProperty, h = Object.assign, Ne = Symbol.for("react.client.reference"), J = Array.isArray, _ = 0, D, ee, re, te, oe, ne, ae;
    p.__reactDisabledLog = !0;
    var z, le, H = !1, V = new (typeof WeakMap == "function" ? WeakMap : Map)(), _e = Symbol.for("react.client.reference"), se, ue = {}, ce = {}, ie = {};
    k.Fragment = Y, k.jsx = function(e, r, t, n, u) {
      return G(e, r, t, !1, n, u);
    }, k.jsxs = function(e, r, t, n, u) {
      return G(e, r, t, !0, n, u);
    };
  }()), k;
}
var ve;
function Ce() {
  return ve || (ve = 1, process.env.NODE_ENV === "production" ? C.exports = ke() : C.exports = Re()), C.exports;
}
var l = Ce();
function Ae({ children: s, variant: b = "primary", onClick: g }) {
  const p = "px-4 py-2 font-semibold rounded-lg transition-all", i = {
    primary: "bg-black text-white hover:bg-blue-700 ",
    secondary: "bg-gray-200 text-black hover:bg-gray-300",
    outline: "border border-blue-600 text-blue-600 hover:bg-blue-100"
  };
  return /* @__PURE__ */ l.jsx("button", { className: `${p} ${i[b]}`, onClick: g, children: s });
}
function Oe() {
  const [s, b] = we(!1);
  return /* @__PURE__ */ l.jsxs("nav", { className: "bg-black text-white p-4", children: [
    /* @__PURE__ */ l.jsxs("div", { className: "container mx-auto flex justify-between items-center", children: [
      /* @__PURE__ */ l.jsx("a", { href: "#", className: "text-xl font-bold", children: "MyUI" }),
      /* @__PURE__ */ l.jsxs("div", { className: "hidden md:flex space-x-6", children: [
        /* @__PURE__ */ l.jsx("a", { href: "#", className: "hover:text-gray-400", children: "Home" }),
        /* @__PURE__ */ l.jsx("a", { href: "#", className: "hover:text-gray-400", children: "About" }),
        /* @__PURE__ */ l.jsx("a", { href: "#", className: "hover:text-gray-400", children: "Services" }),
        /* @__PURE__ */ l.jsx("a", { href: "#", className: "hover:text-gray-400", children: "Contact" })
      ] }),
      /* @__PURE__ */ l.jsx(
        "button",
        {
          className: "md:hidden focus:outline-none",
          onClick: () => b(!s),
          children: s ? "✖" : "☰"
        }
      )
    ] }),
    s && /* @__PURE__ */ l.jsxs("div", { className: "md:hidden flex flex-col space-y-2 mt-4", children: [
      /* @__PURE__ */ l.jsx("a", { href: "#", className: "block py-2 px-4 hover:bg-gray-800", children: "Home" }),
      /* @__PURE__ */ l.jsx("a", { href: "#", className: "block py-2 px-4 hover:bg-gray-800", children: "About" }),
      /* @__PURE__ */ l.jsx("a", { href: "#", className: "block py-2 px-4 hover:bg-gray-800", children: "Services" }),
      /* @__PURE__ */ l.jsx("a", { href: "#", className: "block py-2 px-4 hover:bg-gray-800", children: "Contact" })
    ] })
  ] });
}
function Pe() {
  return /* @__PURE__ */ l.jsx("footer", { className: "bg-black text-white p-6 mt-10", children: /* @__PURE__ */ l.jsx("div", { className: "container mx-auto flex flex-col ", children: /* @__PURE__ */ l.jsxs("p", { className: "text-sm mt-4 md:mt-0 items-center", children: [
    "© ",
    (/* @__PURE__ */ new Date()).getFullYear(),
    " MyUI. All rights reserved."
  ] }) }) });
}
export {
  Ae as Button,
  Pe as Footer,
  Oe as Navbar
};
